Src/Web/Web.Shared.Net/RequestTrackingTelemetryModule.cs

```diff
@@ -102,12 +102,21 @@ public void OnEndRequest(HttpContext context)
                 requestTelemetry.Url = context.Request.UnvalidatedGetUrl();
             }
 
+            if (string.IsNullOrEmpty(requestTelemetry.Context.InstrumentationKey))
+            {
+                // Instrumentation key is probably empty, because the context has not yet had a chance to associate the requestTelemetry to the telemetry client yet.
+                // and get they instrumentation key from all possible sources in the process. Let's do that now.
+                this.telemetryClient.Initialize(requestTelemetry);
```

Comment:
yes it's possible but this is something that I already added for DependencyTelemetry processing on Sergey's suggestion. Need it here as well now. Unfortunately, no way around it and the impact should be small.